### PR TITLE
Add more extensive tests, make build on stable

### PR DESCRIPTION
**Pull request by kingoflolz**

### CODE REVIEW

**Comment by jkilpatr** [comment]
@kingoflolz we probably want to clean this up and land it rather than continuing this big chain of commits. Ideally pulls are kept small otherwise you end up in code review hell.

File: althea_types/src/lib.rs [hunk 1]
```diff
@@ -23,32 +23,184 @@ pub use interop::{Identity, PaymentTx};
...
-    use super::{EthSignature, EthPrivateKey, EthAddress};
 
-    #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    use std::collections::hash_map::DefaultHasher;
+    use std::net::IpAddr;
+    use std::net::Ipv6Addr;
+    use std::hash::{Hash, Hasher};
+    use num256::Uint256;
+
+    use super::*;
```
[the remark this answers is not in the document]
Comment (kingoflolz) [reply althea_types/src/lib.rs]: I think this just imports the things that were defined in this file, but outside of the test module

File: rita/payment_controller/src/lib.rs [hunk 2]
```diff
@@ -153,8 +170,204 @@ impl PaymentController {
...
+            .with_body("bounty OK")
+            .expect(100)
+            .create();
+
+        let (rita_tx, rita_rx) = mpsc::channel();
+
+        let mut pc = PaymentController::new(&new_identity(1));
+
+        for _ in 0..100 {
+            pc.payment_received(new_payment(1), Arc::new(Mutex::new(rita_tx.clone()))).unwrap();
```
[the remark this answers is not in the document]
Comment (kingoflolz) [reply rita/payment_controller/src/lib.rs]: we have to use an arc+mutex to send a "sender" between threads, and payment_received is located in a different thread to the receiver.

File: althea_types/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkilpatr) [reply althea_types/src/lib.rs]: yeha I missed that it was in the test section, my bad.

**Pull request merged**